drivers: ethernet: stm32hal: Fixes source address control

mrjimenez opened the pull request:
The default configuration for all STM32 microcontrollers in HAL is
to replace the ehternet MAC address with the address configured in
replace address zero. In "modules/hal", Grep for:

macDefaultConf.SourceAddrControl = ETH_SOURCEADDRESS_REPLACE_ADDR0;

But this is a bad thing for bridging. Since Zephyr ethernet LL code
always writes the source address, this patch has no impact on normal
operation. And has the benefit of maintaining the source MAC address of
bridged packets.

mrjimenez commented on drivers/ethernet/eth_stm32_hal_v1.c:
```diff
@@ -239,6 +239,11 @@ void eth_stm32_set_mac_config(const struct device *dev, struct phy_link_state *s
+	/* The default configuration is ETH_SOURCEADDRESS_REPLACE_ADDR0,
+	 * which is bad for bridging.
+	 */
+	mac_config.SourceAddrControl = ETH_SOURCEADDRESS_DISABLE;
```
Ok, I have removed these lines from v1.

juickar commented on drivers/ethernet/eth_stm32_hal_v2.c:
```diff
@@ -679,6 +679,12 @@ void eth_stm32_set_mac_config(const struct device *dev, struct phy_link_state *s
+	/* The default configuration is ETH_SOURCEADDRESS_REPLACE_ADDR0,
+	 * which is bad for bridging.
+	 */
+	mac_config.SourceAddrControl = ETH_SOURCEADDRESS_DISABLE;
```
Okay, fair enough let's disable it by default but could you change the comment on top to:

```
	/* Always disable hardware source address replacement.
	 * Zephyr network stack sets the source MAC address and
	 * therefore hardware replacement should not be enabled.
	 */
```
or something similar

mrjimenez committed:
drivers: ethernet: stm32hal: Fixes source address control

The default configuration for all STM32 microcontrollers in HAL is
to replace the ehternet MAC address with the address configured in
replace address zero. In "modules/hal", Grep for:

macDefaultConf.SourceAddrControl = ETH_SOURCEADDRESS_REPLACE_ADDR0;

But this is a bad thing for bridging. Since Zephyr ethernet LL code
always writes the source address, this patch has no impact on normal
operation. And has the benefit of maintaining the source MAC address
of bridged packets.

Signed-off-by: Marcelo Roberto Jimenez <marcelo.jimenez@gmail.com>
